Mathlib/RingTheory/DedekindDomain/Ideal.lean

```diff
@@ -1536,4 +1533,41 @@ theorem multiplicity_normalizedFactorsEquivSpanNormalizedFactors_symm_eq_multipl
     multiplicity_normalizedFactorsEquivSpanNormalizedFactors_eq_multiplicity hr ha]
 #align multiplicity_normalized_factors_equiv_span_normalized_factors_symm_eq_multiplicity multiplicity_normalizedFactorsEquivSpanNormalizedFactors_symm_eq_multiplicity
 
+/-- The bijection between the set of prime factors of the ideal `⟨r⟩` and the set of prime factors
+  of `r` preserves `count` of the corresponding multisets. See
+  `multiplicity_normalizedFactorsEquivSpanNormalizedFactors_eq_multiplicity` for the version
+  stated in terms of multiplicity. -/
+theorem count_span_normalizedFactors_eq {r X : R} (hr : r ≠ 0) (hX : Prime X) :
+    Multiset.count (Ideal.span {X} : Ideal R) (normalizedFactors (Ideal.span {r}))  =
+        Multiset.count (normalize X) (normalizedFactors r) := by
+  have := multiplicity_eq_multiplicity_span (R := R) (a := X) (b := r)
+  rw [multiplicity_eq_count_normalizedFactors (Prime.irreducible hX) hr,
+    multiplicity_eq_count_normalizedFactors (Prime.irreducible ?_), normalize_apply,
+    normUnit_eq_one, Units.val_one, one_eq_top, mul_top, Nat.cast_inj] at this
+  simp only [normalize_apply, this]
+  · simp only [Submodule.zero_eq_bot, ne_eq, span_singleton_eq_bot, hr, not_false_eq_true]
+  · simpa only [prime_span_singleton_iff]
```

Comment:
```suggestion
  have hIrred : Irreducible (span {X}) := Prime.irreducible <| prime_span_singleton_iff.mpr hX
  have hneq0 : span {r} ≠ 0 := by simp only [Submodule.zero_eq_bot, ne_eq, span_singleton_eq_bot,
    hr, not_false_eq_true]
  have := multiplicity_eq_multiplicity_span (R := R) (a := X) (b := r)
  rw [multiplicity_eq_count_normalizedFactors (Prime.irreducible hX) hr,
    multiplicity_eq_count_normalizedFactors hIrred hneq0, normalize_apply,
    normUnit_eq_one, Units.val_one, one_eq_top, mul_top, Nat.cast_inj] at this
  simp only [normalize_apply, this]
```
I feel it's confusing when `rw` introduces extra goals. Adding some `have`s beforehand hopefully makes it clearer. Also `r ≠ 0 → span {r} ≠ 0` should perhaps be a lemma in mathlib?